lib/urlapi.c

```diff
@@ -797,92 +817,90 @@ UNITTEST int dedotdotify(const char *input, size_t clen, char **outp)
   outptr = out;
 
   do {
-    bool dotdot = TRUE;
-    if(*input == '.') {
+    if(is_dot(&input, &clen)) {
```

Comment:
This can only happen at the start of the buffer (and will misbehave otherwise).  Should there be an assertion?

Reply:
I figured the testing would be enough, but yes an assert is always helpful. Maybe I can even move out this logic from the loop and only do it once. I'll experiment a little. Thanks!

Reply:
that actually made the logic better and easier to read...